packages/cinterop/src/darwin/kotlin/io/realm/kotlin/internal/interop/RealmInterop.kt

```diff
@@ -208,6 +211,12 @@ fun realm_value_t.set(memScope: MemScope, realmValue: RealmValue): realm_value_t
                 }
             }
         }
+        is ByteArray -> {
+            type = realm_value_type.RLM_TYPE_BINARY
+            (0 until value.size).map {
+                binary.data?.set(it, value[it].toUByte())
```

Comment:
Seems that a alloc array is missing here.

Reply:
Double checked parameterised queries and they failed hard 🙈 I fixed it and wrote a test